### ISSUE TITLE
Adding my own m3u playlists with EPG

### ISSUE THREAD

**Issue by CHJ85** [MythTV/mythtv#519]
Hi there. I haven't been able to find many guides (let alone youtube videos) about mythtv. So I was looking around trying to find out how I can add m3u playlists, as I'd like to use mythtv for steaming services such as Pluto TV, LocalBTV and STiRR as I only watch TV through apps. And I think this is the direction mythtv needs to go in order to be relevant today. Because MythTV itself is a great app. It's just not needed for a lot of people in a state that it's in right now by only supporting TV antennas and cable boxes. This is not something that people really use anymore unless it comes bundled with your internet or something. So that's when I was thinking that maybe you guys should borrow a page from Channels PVR, Stremium and SelectTV. The two latter ones being cloud based DVR. But they allow you to add live tv streaming services such as Sling TV, Fubo TV and Hulu as well as your own m3u and epg. And then I look at Hypnotix from LinuxMint. And I'm like "If this app came with an EPG and the ability PVR functionality, it'd be golden".
And this is where mythtv comes in. MythTV has all those features, but lacks the ability to add m3u playlists and third party streaming services.
I think people would be using MythTV if it did.

**Comment by glenb77** [comment]
Myth does support playlists through the 'freebox' tuner. However you are on your own how to get epg data to it. But i use it regularly with m3u lists and made customized epg with schedules direct listings here in usa. This wiki is more or less the way:
https://www.mythtv.org/wiki/User_Manual:Setting_up_HTTP_Live_Streaming_Recorder
Personally i have found that setting up tvheadend for my m3u services and then have myth record from the tvheadend url is a better solution for me.

**Comment by CHJ85** [comment]
@glenb77 Right. But like I said, that Wiki is extremely outdated. The GUI is completely different and I have no idea where these settings are.

**Comment by glenb77** [comment]
the mythtv-setup is pretty much the same as the wiki shows. first create an m3u list. Here is a single channel entry from my m3u file for cnn:

#EXTINF:0,729 - CNN
#EXTMYTHTV:xmltvid=I37.10142.zap2it.com
http://glen:g@xxxxx.net:9981/stream/channelid/1582193683?profile=pass

so here, this is going to be channel 729, it will be named 'CNN'. the next line #extmythtv:xmltivid= is where you put your listings source, and the last line is the actual http url. again i use tvheadend so thats my link but it is where the url goes.

when you have that list done. then go to mythtv-setup and follow along in the wiki by setting a capture card, naming a video source and then importing the m3u list to myth via the scan in input connections menu. 

I then run mythfilldatabase with my source xml file of listings and specify the group and the listings update and it is then incorporated to the guide and ready to record when needed

i don't know if i can put my email here so i am not, but i am willing to help you along, but i don't think this is the right place. so if you want the help and can give me an email somehow, we can go further.

**Comment by CHJ85** [comment]
Thanks but I think I'll pass.
This is a bit over my pay grade.
For starters, it says it's missing a database when I open up mythtv setup.
And according to the wiki and yourself, I have to use pre-generated m3u. So in other words, I am limited in terms of what channels I can watch. I wanted to use it for Sling TV, which means that the m3u playlist would need to be generated every 30 second because that's when the token expires.

**Comment by glenb77** [comment]
On myth-setup likely you need to follow wiki to set up database for first time install. On the iptv, I agree that from a development standpoint mythtv should focus more on iptv. This is not an issue with mythtv not working. if you continue with mythtv you should use the forum at mythtv.org or the mail list. I'm not familiar with how sling delivers it service, but if it is hls based with a changing token going into the url, that is possible through a proxy, but yes it would be complicated in myth or anything else for that matter. I did it using tvheadend on something with a similar approach. This issue should be closed as it is not an issue.

**Comment by garybuhrmaster** [comment]
On Fri, Mar 11, 2022 at 5:03 PM CHJ85 ***@***.***> wrote:

> And this is where mythtv comes in. MythTV has all those features, but lacks the ability to add m3u playlists and third party streaming services.

As others have pointed out, MythTV does support
m3u and IPTV (even if you feel it is too daunting to
deal with).  However, for most streaming services,
things get far far more complicated if you intend to
try to "record" such content (rather than just run
the streaming app itself inside some instantiated
browser window), as even when you can extract
the m3u from the service, it is often a rotating value,
which you need to refresh regularly, and in addition
a number of those services use widevine
protection (for at least some content), which
requires one to instantiate a (usually) headless
chromium browser to be able to extract the stream.
And some of those same services use a licensed
authentication scheme that the IP owner will (and
has) gone after even free software for using the
IP without a license (that is one reason why at
least some of the streaming services you
mentioned have a required charge for some of
their more advanced features, as they have to
pay the IP owner(s) for the licenses).

"It's worse than you think."
"It usually is."
